### DISCUSSION TITLE
Mapping property Odata

### DISCUSSION THREAD

**Discussion by Paolo-lab** [iTwin/itwinjs-core#7243] [Q&A] [answered]
Hello everyone, we have developed for our end customer the path related to the odata, as a connector they were using was deprecated.
Everything works and for this I am grateful, but the customer has difficulty mapping all the heads of the properties
each element as it was structured by the customer has in itself
an instance parameter and a type parameter
I attach a small image of a span of a bridge with the two distinct properties:
<img width="242" alt="img_odata1" src="https://github.com/user-attachments/assets/574b6af7-a56b-46f0-8c08-5a90430cc158">

While it has no problems mapping the properties inside Selected Item, and therefore the odata are populated correctly, it is unable to capture the IfcBeamType properties even if this property is listed during the mapping, I attach an image:
<img width="445" alt="img_odata2" src="https://github.com/user-attachments/assets/1b17346b-1ccd-4c6a-9edd-1deb1dd378eb">

The property then comes out in the list of odata but is always empty. I attach an image:

<img width="900" alt="img_odata3" src="https://github.com/user-attachments/assets/495f1675-e46e-495b-ac81-7df945a80aa2">

so I was wondering if we were somehow working badly, or we did not understand well the features made available to us

Thank you for the help, without it I would be lost.
Paul

**Comment by Guillar1** [comment]
Hello,

If no value is returned, it typically indicates that the property is either `null` or empty in the actual data. The listed properties only reflects possible properties for the `Group` query, but it does not guarantee every row in the resulting extraction will contain a value. 

The first way to check if a value is present is through the Properties widget in your first screenshot. By selecting an element within the Group in the viewer, you can see whether the element contains a value. If no value is displayed, it is likely `null` or empty in the actual extraction.

If it is still unclear whether a specific property contains a value in any of the resulting rows you can run the query in [iModelConsole](https://imodelconsole.bentley.com/). Be aware that if the Group was created through the UI, you will have to adjust the `SELECT` clause to see all the columns, the default `SELECT` clause only selects the `ECInstanceId` and `ECClassId` columns.

**Reply by Paolo-lab** [reply]
Hello Thanks for the reply, the data is present, I attach an image:
<img width="952" alt="img_odata4" src="https://github.com/user-attachments/assets/f628ba50-b90b-487f-aa7c-08dfb169e642">

as you can see by selecting the odata I opened the properties and while in the list of odata for this specific element it is empty, in the properties the value exists.

I also did some checks on the console by directly calling
select * from IFCDynamic.ifcAspect_Identity_Data and the data is there

To create the group the customer uses "Query Generation Tool-> Selection" so for example selects a beam and then offers it the two properties divided in the same way as the properties of the elements. , it counts the number of elements extracted. after which I map the individual properties that I want in my odata, this is more or less the procedure that follows.

Thanks Paul

**Reply by Guillar1** [reply]
I understand now, I will get back to you after I discuss this with the team.

**Reply by Guillar1** [reply, marked as answer]
Hello @Paolo-lab,

Upon investigation with a similar iModel, we found that the `OmniClass Number` property shown in the `Properties` widget comes from the aspect of the `TypeDefinition` of the selected element, not from the aspect of the selected element itself. The `Properties Selection` menu is targeting the aspect of the selected element, which does not have that property set. Unfortunately, the service and widget do not yet support navigating to a related instance's aspect. We’ve added this to our backlog for prioritization.

As a workaround, you can manually create a new Group and write the query manually for the `TypeDefinition` table and fetch the OmniClass Number through there.
